search/request.go

```diff
@@ -0,0 +1,179 @@
+/*
+ * Copyright 2022 Tigris Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package search
+
+import (
+	"encoding/json"
+	"github.com/tigrisdata/tigris-client-go/driver"
+	"github.com/tigrisdata/tigris-client-go/fields"
+	"github.com/tigrisdata/tigris-client-go/filter"
+)
+
+// Request for search
+type Request struct {
+	Q            string
+	SearchFields []string
+	Filter       filter.Filter
+	Facet        *FacetQuery
+	ReadFields   fields.Read
```

Comment:
are these fields used by the read API as well? If yes, then we need to have different Readfield's for Search so that there is no dependency.